core/http-auth-aws-event-stream/src/main/java/software/amazon/awssdk/http/auth/aws/eventstream/internal/TrailingDataFramePublisher.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.aws.eventstream.internal;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+
+/**
+ * A publisher which wraps a publisher and appends a trailing frame once the wrapped publisher is finished.
+ */
+@SdkInternalApi
+public final class TrailingDataFramePublisher implements Publisher<ByteBuffer> {
```

Comment:
I'll need to take a deeper look at this class, but we need tck tests for reactive streams implementations. https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md#reactive-streams

Reply:
We need TCK test.https://github.com/reactive-streams/reactive-streams-jvm/tree/master/tck